exporter/googleservicecontrolexporter/factory.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package googleservicecontrolexporter
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configretry"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+	"google.golang.org/api/impersonate"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/google"
+	"google.golang.org/grpc/credentials/oauth"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "googleservicecontrol"
+)
+
+var (
+	// 16s is the Service Control API default:
+	// https://github.com/googleapis/googleapis/blob/d68746128bbb1c5729ff97132f8532e36f796929/google/api/servicecontrol/v1/servicecontrol_grpc_service_config.json#L26
+	// It is important to keep it as is: go/slm-monitoring-opentelemetry-batching.
+	defaultTimeout  = 16 * time.Second
+	defaultEndpoint = "servicecontrol.googleapis.com:443"
+	clientProvider  = New
+)
+
+// Config defines configuration for Service Control Exporter
+type Config struct {
+	ServiceName               string `mapstructure:"service_name"`
+	ConsumerProject           string `mapstructure:"consumer_project"`
+	ServiceControlEndpoint    string `mapstructure:"service_control_endpoint"`
+	ServiceConfigID           string `mapstructure:"service_config_id"`
+	ImpersonateServiceAccount string `mapstructure:"impersonate_service_account"`
+	// Whether to use servicecontrol library or raw sc client.
+	// Defaults to `true`, so that existing customers are unaffected by changes.
+	// See go/agent-gdce
+	UseRawServicecontrolClient string `mapstructure:"use_raw_sc_client"`
```

Comment:
[nit] What is the difference here? When would a new user want to do the opposite?